AxonDeepSeg/apply_model.py

```diff
@@ -1,493 +1,46 @@
-    :param segmentations_filenames: List of the names of the segmentations files, to be used when creating the files.
-    :param inference_batch_size: Size of the batches fed to the network.
-    :param overlap_value: Int, number of pixels to use for overlapping the predictions.
-    :param resampled_resolutions: List of the resolutions (in µm) to resample to.
-    :param acquired_resolution: List of the resolutions (in µm) for native images.
-    :param prediction_proba_activate: Boolean, whether to compute probability maps or not.
-    :param write_mode: Boolean, whether to create segmentation images or not.
-    :param gpu_per: Percentage of the GPU to use, if we use it.
-    :param verbosity_level: Int, level of verbosity. The higher, the more information is displayed.
-    :return: List of predictions, and optionally of probability maps.
-    """
-
-    # If string, convert to Path objects
-    path_acquisitions_folders = convert_path(path_acquisitions_folders)
-    path_model_folder = convert_path(path_model_folder)
-    # Processing input so they are lists in every situation
-    path_acquisitions_folders, acquisitions_filenames, resampled_resolutions, segmentations_filenames = \
-        list(map(ensure_list_type, [path_acquisitions_folders, acquisitions_filenames, resampled_resolutions,
-                                    segmentations_filenames]))
-
-    if len(segmentations_filenames) != len(path_acquisitions_folders):
-        segmentations_filenames = [str(axonmyelin_suffix)] * len(path_acquisitions_folders)
-
-    if len(acquisitions_filenames) != len(path_acquisitions_folders):
-        acquisitions_filenames = ['image.png'] * len(path_acquisitions_folders)
-
-    if len(resampled_resolutions) != len(path_acquisitions_folders):
-        resampled_resolutions = [resampled_resolutions[0]] * len(path_acquisitions_folders)
-
-    # Generating the patch to acquisitions and loading the acquisitions resolutions.
-    path_acquisitions = [path_acquisitions_folders[i] / e for i, e in enumerate(acquisitions_filenames)]
-
-    # If we did not receive any resolution we read the pixel size in micrometer from each pixel.
-    if acquired_resolution == None:
-        if (path_acquisitions_folders[0] / 'pixel_size_in_micrometer.txt').exists():
-            resolutions_files = [open(path_acquisition_folder / 'pixel_size_in_micrometer.txt', 'r')
-                                 for path_acquisition_folder in path_acquisitions_folders]
-            acquisitions_resolutions = [float(file_.read()) for file_ in resolutions_files]
-        else:
-            exception_msg = "ERROR: No pixel size is provided, and there is no pixel_size_in_micrometer.txt file in image folder. " \
-                            "Please provide a pixel size (using argument -s), or add a pixel_size_in_micrometer.txt file " \
-                            "containing the pixel size value."
-            raise Exception(exception_msg)
```

Comment:
I think this part should stay in the axon_segmentation function so we can extract the pixel size from the txt file using the 00-notebook (I left a similar comment there to).

Reply:
Done in e785e81, hope it works as intented.